docs/user-guide/configmap.md

```diff
@@ -0,0 +1,422 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# ConfigMap
+
+The ConfigMap API resource holds configuration data in a string-to-string map.  ConfigMaps can be
+consumed in pods similarly to [Secrets](secrets.md): in environment variables or via volume plugin.
+
+
+## Overview of ConfigMap
+
+The ConfigMap API resource holds configuration data that can be consumed in pods or used to store
+configuration data for system components such as controllers.  ConfigMap is similar to
+[Secrets](secrets.md), but designed to more conveniently support working with strings that do not
+contain sensitive information.
+
+Both users and system components may store configuration data in ConfigMap.
+
+Configuration data can be consumed in a variety of ways.  ConfigMaps can be used to:
+
+1.  Populate the value of environment variables
+2.  Set command-line arguments in a container
+3.  Populate a volume
+
+### Restrictions
```

Comment:
Maybe we move this section below the tutorial? If you provide a TOC at the top it will be easy to find the important information without pushing it above the meat of the docs.